### DISCUSSION TITLE
Option to disable / enable Armoury Crate Service?

### DISCUSSION THREAD

**Discussion by crabnebula007** [seerge/g-helper#2542] [Ideas]
I have a G14 2023 and want to use G-Helper while leaving Armoury Crate installed so that I can use it occasionally to control my ROG keyboard.

However, this leads to two problems:

- Every time I change my GPU Mode in G-Helper, my performance mode settings are reset to Armoury Crate's current default.
- As soon as I enable the dGPU (eg, switch from Optimized to Normal mode), the screen brightness slider stops working immediately  and I have to remove and reinstall the dGPU in Device Manager to fix this.

I've found that stopping the Windows service "Armoury Crate Service" prevents both of these issues and allows a fully functional G-Helper without removing Armoury Crate.

Therefore, I was wondering if it would be feasible to add an option to disable / enable the Armoury Crate Service, similar to how the app handles Asus services (but as a separate setting). Or even better, an option to automatically disable the service when G-Helper launches and restart it when manually quitting the app with the Quit button. That way, whenever G-Helper is running, AC would automatically be disabled to prevent any interference.

Thanks!

**Comment by seerge** [comment]
@crabnebula007 you can try to add an optional parameter to ``%appdata%\GHelper\config.json``

```
"stop_ac" : 1,
```

For this build
[GHelper.zip](https://github.com/seerge/g-helper/files/15210916/GHelper.zip)

**Reply by crabnebula007** [reply]
Thank you! Unfortunately it doesn't seem to have any effect. Not sure if you use the service name or the display name when disabling/enabling it, but perhaps it is because I didn't capitalize properly (sorry). The correct display name is: "ARMOURY CRATE Service"

![image](https://github.com/seerge/g-helper/assets/15935837/56562d36-a8a1-4ae1-a66a-2087ab172980)

**Reply by seerge** [reply]
@crabnebula007 service name is entered correctly
https://github.com/seerge/g-helper/blob/8f2a13b1adea1a2fc0ee5cd8831e3a5c0d4e1d26/app/Helpers/OptimizationService.cs#L28

Can you post a log here (after clicking STOP)?

**Reply by crabnebula007** [reply]
Sorry, I thought you had programmed it to disable the AC services when launching G-Helper, but by looking at the code I understood it was tied to the existing Stop/Start button for Asus Services.

And yes, it does work.

However, I also noticed in the code there is an error in the StartAsusServices function, it is checking for AppConfig.IsAlly instead of AppConfig.IsStopAC before enabling the services.

Out of curiosity, is there a reason you're setting the AC services to Manual instead of Disabled?

**Reply by seerge** [reply]
@crabnebula007 i have already fixed that, and build above is updated. 

It sets services to Manual instead of Disabled, so when you launch AC - it can just start them by itself (this is how it was already done for ROG Ally)

**Reply by crabnebula007** [reply]
Thank you very much. It works well!
